R/helper_functions.R

```diff
@@ -790,21 +790,20 @@ check_samplr_arguments <-
   }
 
 
-clean_condition_names <- 
-  function(assignment, conditions) {
-    if (is.factor(conditions)) {
-      return(factor(assignment, levels = levels(conditions)))
-    }
-    
-    if (is.numeric(assignment)) {
-      return(assignment)
-    }
-    
-    if (is.logical(assignment)) {
-      return(as.numeric(assignment))
-    }
-    return(factor(assignment, levels = conditions))
+clean_condition_names <- function(assignment, conditions) {
+  if (is.factor(conditions)) {
+    return(factor(assignment, levels = levels(conditions)))
+  }
+  
+  if (is.numeric(conditions)) {
+    return(assignment)
   }
+  
+  if (is.logical(conditions)) {
+    return(as.numeric(assignment))
```

Comment:
If people gave complete_ra(10, conditions = c(T, F)), I think they'd want back Ts and Fs, no?

Reply:
That makes sense.